pkg/jobs/cleanup_missing_domains.go

```diff
@@ -0,0 +1,37 @@
+package jobs
+
+import (
+	"context"
+
+	"github.com/content-services/content-sources-backend/pkg/dao"
+	"github.com/content-services/content-sources-backend/pkg/db"
+	"github.com/content-services/content-sources-backend/pkg/pulp_client"
+	"github.com/rs/zerolog/log"
+)
+
+func CleanupMissingDomains() {
+	err := db.Connect()
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to connect to database")
+	}
+
+	daoReg := dao.GetDaoRegistry(db.DB)
+	ctx := context.Background()
+
+	domains, err := daoReg.Domain.List(ctx)
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to list domains")
+	}
+	for _, domain := range domains {
+		pulpHref, err := pulp_client.GetGlobalPulpClient().LookupDomain(ctx, domain.DomainName)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to lookup pulp domain")
+		}
```

Comment:
it'd be better if we explicitly checked for a 404, in case some other error occured.

Reply:
it looks like we don't return an error if the domain we filter by isn't found. should [this](https://github.com/content-services/content-sources-backend/blob/79e0a48e59ef3388f8f295050201da4aac04107f/pkg/pulp_client/domains.go#L86) return a 404 if the domain isn't found?

Reply:
ohh i see, it returns empty string.  Thats fine, i would just ignore the pulpHref if an err is thrown and skip to the next one?